target/linux/realtek/files-5.10/arch/mips/rtl838x/setup.c

```diff
@@ -50,24 +91,19 @@ void __init plat_mem_setup(void)
 
 void __init plat_time_init(void)
 {
-	struct device_node *np;
-	u32 freq = 500000000;
+	u32 cpu, lxb, mem;
 
 	of_clk_init(NULL);
 	timer_probe();
 
-	np = of_find_node_by_name(NULL, "cpus");
-	if (!np) {
-		pr_err("Missing 'cpus' DT node, using default frequency.");
-	} else {
-		if (of_property_read_u32(np, "frequency", &freq) < 0)
-			pr_err("No 'frequency' property in DT, using default.");
-		else
-			pr_info("CPU frequency from device tree: %dMHz", freq / 1000000);
-		of_node_put(np);
-	}
+	plat_get_frequencies(&cpu, &lxb, &mem);
```

Comment:
arg, and now I see also that this is arch specific setup code, well at still a good improvement :)

Does mips not use normal drivers/clk/rtl838x?

Reply:
OpenWrt isn't on MIPS_GENERIC yet, so we can't really use this upstream. How are early clock dependencies for CPUs normally handled? I'd imagine at least the code for `fixed-clock` needs to allow for this (since I use this with the MIPS_GENERIC kernel), so maybe that can provide some hints on how this can be done without making it platform-specific.

Reply:
rewritten